plugins/quota/actor.go

```diff
@@ -0,0 +1,89 @@
+package quota
+
+import (
+	"os"
+	"path"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	log "github.com/sirupsen/logrus"
+)
+
+type quotadActor struct{}
+
+const (
+	quotaDaemonKey = "enable"
+)
+
+// isQuotadAffectedNode returns true if there are local bricks of volume on
+// which quota is enabled
+func isQuotadAffectedNode(volumes []*volume.Volinfo) bool {
+	for _, v := range volumes {
+		if !isQuotaEnabled(v) {
+			continue
+		} else if v.State != volume.VolStarted {
+			continue
+		} else {
+			bricks := v.GetLocalBricks()
+			if len(bricks) > 0 {
+				return true
+			}
+			continue
+		}
+	}
+	return false
+}
+
+func (actor *quotadActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
+	if key != quotaDaemonKey {
+		return nil
+	}
+	quotadDaemon, err := NewQuotad()
+	if err != nil {
+		return err
+	}
+	// Create pidfile dir if not exists
+	if err := os.MkdirAll(path.Dir(quotadDaemon.pidfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	// Create logFiledir dir
+	if err := os.MkdirAll(path.Dir(quotadDaemon.logfilepath),
+		os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	volumes, err := volume.GetVolumes()
+	if err != nil {
+		logger.WithError(err).Error("Failed to get volumes")
```

Comment:
question:
In case of failure here, do we need to delete ```pidfile dir```?

Reply:
Not necessary. the pidfile dir is not used for any comparison (only the pidfile itself is used to check if the process is there or not). And failed to get volumes can happen for some reason that is irrelevant to the quotad running. in case if it was already running, the pidfile dir will have the pidfile and deleteing the dir is not the thing to do by then.